src/main/java/io/jenkins/plugins/gitlabserver/servers/GitLabServer.java

```diff
@@ -140,23 +141,46 @@ public String getCredentialsId() {
     /**
      * Data Bound Constructor
      *
-     * @param name          A unique name to use to describe the end-point, if empty replaced with a random name
-     * @param serverUrl     The URL of this GitLab Server
-     * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
      * @param credentialsId The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for
-     *                      auto-management of hooks.
+     *                      GitLab Server Authentication to access GitLab APIs
      */
     @DataBoundConstructor
-    public GitLabServer(@Nonnull String name, @NonNull String serverUrl, boolean manageHooks,
-                        @CheckForNull String credentialsId) {
-        this.manageHooks = manageHooks;
+    public GitLabServer(@CheckForNull String credentialsId) {
         this.credentialsId = credentialsId;
```

Comment:
I don't think credentials id is mandatory if the repo is public.

Reply:
Okay. Just wondering if GitHub Plugin did it wrongly?